docs/src/tutorials/calibration.qmd

```diff
@@ -0,0 +1,94 @@
+
+## Notation
+We are given a labeled dataset $x_t, y_t \in X \times Y$ for $t = 1, 2, ..., T$ of i.i.d. realizations of random variables $X, Y \sim P$, where $P$ is the data distribution. 
+Given $x_t$, a forecaster $H : X \rightarrow (Y \rightarrow [0, 1])$ outputs at each step $t$ a CDF $F_t(y)$ targeting the label $y_t$. When $Y$ is continuous, $F_t$ is a cumulative probability distribution (CDF).  We will use $F^{−1}_t: [0, 1] → Y$ to denote the quantile function $F^{−1}_t (p) = inf\{y : p ≤ F_t(y)\}$.
+
+## Calibration in the Regression case
+In the regression case, we say that the forecaster H is  (on average) calibrated if
+
+$$\frac{\sum_{t=1}^T \mathbb{1} \{ y_t \leq F_t^{-1}(p)  \} }{T} \rightarrow p \quad \text{for all}\quad  p \in [0,1]$$
+
+as $T \rightarrow \infty$.\  In other words, the empirical and the predicted CDFs should match as the dataset size goes to infinity.
+Perfect Calibration is a sufficient condition for average calibration, the opposite however is not necessarily true: a model can be average calibrated but not perfectly calibrated.\
+From now on when we talk about calibration we will implicitly talk about average calibration rather than perfect calibration. 
+
+
+### Sharpness 
+Calibration by itself is not sufficient to produce a useful forecast. For example, it is easy to see that if we use for the forecast the marginal distribution $F(y) = \mathbb{P}(Y ≤ y)$, without considering the input feature $X$, the forecast will be calibrated but still not accurate. In order to be useful, forecasts must also be sharp, which,in a regression context, means that the confidence intervals should all be as tight as possible around a single value. More formally, we want the variance $var(F_t)$ of the random variable whose CDF is $F_t$ to be small.\
+As  a sharpness score of the forecaster, Kuleshov et al. proposed the average predicted variance
+
+$$sharpness(F_{1} ,\dots, F_T) = \frac{1}{T} \sum_{t=1}^T var{F_t}$$ \
+the smaller the sharpness, the tighter will be the confidence intervals on average.
+
+### Calibration Plots 
+
+To check the level of calibration, Kuleshov et al. proposed a calibration plot that display the true frequency of points in each confidence interval relative to the predicted fraction of points in that interval.\
+More formally, we choose $m$ confidence levels $0 ≤ p_1 < p_2 < . . . < p_m ≤ 1$; for each threshold $p_j$ , and compute the empirical frequency
+
+$$\hat{p}_j = \frac{|\{ y_t|F_t(y_t) \leq p_t, t= 1,2,\dots,T    \} |}{T}.$$
+
+To visualize the level of average calibration, we plot $\{(p_j,\hat{p_j}) \}_{j=1}^M$; A forecaster that is calibrated  will correspond a straight line on the plot that goes from $\{0,0\}$ to $\{1,1\}$ . \
+As a measure of the level of miscalibration of the forecaster, differently from what it is suggested in the original paper, we a measure  the area between the bisector of the plot and the line produced by the forecaster.
```

Comment:
"we measure"

Might need to elaborate a bit here on why we do this differently and how exactly the area is computed/measured.

Reply:
it's not really different, more like a simplification  ![pic](image.png) . he uses the rectangular rule with variable weights, i use the trapezoidal rule.